### DISCUSSION TITLE
secp256k1 encryption - error

### DISCUSSION THREAD

**Discussion by mspencer** [openpgpjs/openpgpjs#1806] [Q&A] [answered]
Hi

I have given a test public key, it looks like:
**-----BEGIN PGP PUBLIC KEY BLOCK-----
Version: BCPG v1.62

mE8EZy1dBhMFK4EEAAoCAwTjXEozIohEbAUSGHjTeTL9ArRUPxjnHxlo6HDhn+Gy
........
VcN9u8HTiahg8iJQ4SopNovGXFAxAQCDOM4nP3GrXS03oJqNY+FNgGN5lPuXujw9
SfBD6Q7rBg==
=43dW
-----END PGP PUBLIC KEY BLOCK-----**
I cannot change anything about the key.

When I tried to encrypt using opengpg 6.0.0 module in Node, I use this code:
**const encryptedCardDetails = await openpgp.encrypt({
      message: await openpgp.createMessage({ text: 'Hello, World!' }),
      encryptionKeys: publicKey
 });**

But got the error:
**"Error encrypting message: Could not verify primary key: Support for ecdsa keys using curve secp256k1 is disabled."**

How can I get a successful encryption please?

Thanks!

**Comment by larabr** [comment, marked as answer]
That specific curve is[ rejected by default](https://github.com/openpgpjs/openpgpjs/blob/main/src/config/config.js#L294-L299) since it hasn't been standardized in RFC9580.

You can pass `config: { rejectCurves: new Set() }` when encrypting, or change the global configuration default with  `openpgp.config.rejectCurves = new Set()`;

**Reply by mspencer** [reply]
Thanks a lot, @larabr 

After clear the openpgp rejectCurves in the config object using `openpgp.config.rejectCurves.clear();`, I can proceed on with the secp256k1 encryption. :)

**Reply by larabr** [reply]
Ah yes, apologies, it's indeed of type `Set`, not `Array` 🙂 
I also updated my original answer accordingly, for future reference.
